crates/language/src/language_settings.rs

```diff
@@ -35,13 +36,75 @@ pub fn init(cx: &mut AppContext) {
 }
 
 /// Returns the settings for the specified language from the provided file.
-pub fn language_settings<'a>(
+pub fn language_settings(
     language: Option<&Arc<Language>>,
     file: Option<&Arc<dyn File>>,
-    cx: &'a AppContext,
-) -> &'a LanguageSettings {
+    cx: &AppContext,
+) -> LanguageSettings {
```

Comment:
We do not have to allocate in case we either fail to locate the editorconfig (a lot of projects including ours) or the merge result being the same as before.

Let's use `Cow<LanguageSettings>` as a return value to make it better?

Reply:
This seems reasonable, I'm just a nooby but I'll see how it is done! I guess there are other examples I can find in the codebase!